relay-server/src/actors/project_buffer.rs

```
@@ -290,55 +338,26 @@ impl BufferService {
-                                    .push_bind(v.0.sampling_key.to_string())
-                                    .push_bind(envelope_bytes);
-                            }
-                            Err(err) => {
-                                relay_log::error!(
-                                    "failed to serialize the envelope: {}",
-                                    LogError(&err)
-                                )
-                            }
-                        });
-
-                        let query = query_builder.build();
-                        if let Err(err) = query.execute(&db).await {
-                            relay_log::error!(
-                                "failed to buffer envelopes to disk: {}",
-                                LogError(&err)
-                            )
-                        }
-                        // Reset the builder to initial state set by `QueryBuilder::new` function,
-                        // so it can be reused for another chunk.
-                        query_builder.reset();
-                    }
-                });
+                    // A RESERVED lock will be acquired when the first INSERT, UPDATE, or DELETE statement is executed.
+                    // Once we started spooling our in-memory buffer, the dequeue from the disk
+                    // will be blocked till this operations either fails, or commits.
+                    sqlx::query("BEGIN").execute(&mut connection).await?;
+
+                    Self::do_spool(&mut connection, buf).await?;
+
+                    // The SQL command "COMMIT" does not actually commit the changes to disk. It just turns autocommit back on.
+                    // Then, at the conclusion of the command, the regular autocommit logic takes over and causes the actual commit to disk to occur.
+                    // TODO: re-try failed commit?
+                    sqlx::query("COMMIT").execute(&mut connection).await?;
+                }
```

Is it ok to do nothing if the connection cannot be acquired?

---

IMO `try_spool` should return an error in that case. If I call `try_spool` and it doesn't return an error, I'd expect I spooled. If there's any inner detail (i.e. connection could not be acquired), I'd expect I get the error so that I can choose to retry again if I want to.